typing/typecore.ml

```diff
@@ -680,15 +680,16 @@ end) = struct
                    usage lid env expected_type lbls =
     let scope = match scope with None -> lbls | Some l -> l in
     let warn_if_ambiguous lbl rest =
-      Printtyp.Conflicts.reset ();
```

Comment:
Are you sure it's OK to push that call down? `ambiguous_types` does call `Printtyp` in some situations.
I guess @Octachron should have a look at this, as owner of the "conflicts" infrastructure.

Reply:
The function `Printtyp.Conflicts.reset ()` should indeed be called before `ambiguous_types` to start the conflict recording in `ambiguous_types` from a certainly clean state (or not at all under the supplementary assumption that all other users of conflicts cleaned the state after use) . Calling `Printtyp.Conflicts.reset` between `ambiguous_types` and `print_conflicts` erases any conflict explanations before they are printed.
I will send a test that catches this difference in behavior.

Reply:
This is what I suspected, but I couldn't find an issue with my own manual inspection of the code, so I made the change in a commit marked for "separate review". I will get rid of the commit now.